packages/client/src/object/fetchPage.ts

```diff
@@ -122,11 +125,17 @@ export type FetchPageResult<
   L extends ObjectOrInterfacePropertyKeysFrom2<Q>,
   R extends boolean,
 > = Promise<
-  PageResult<
-    ObjectOrInterfacePropertyKeysFrom2<Q> extends L
-      ? (DefaultToFalse<R> extends false ? Osdk<Q> : Osdk<Q, "$all" | "$rid">)
-      : (DefaultToFalse<R> extends false ? Osdk<Q, L> : Osdk<Q, L | "$rid">)
-  >
+  FetchPageWithErrorsResult<Q, L, R>
+>;
+
+export type FetchPageWithErrorsResult<
```

Comment:
Should we call this `FetchPageResult` like it was before but drop the `Promise<>` part? That way you can compose like below?